azurerm/resource_arm_storage_share_test.go

```diff
@@ -223,32 +297,107 @@ func testCheckAzureRMStorageShareDestroy(s *terraform.State) error {
 }
 
 func testAccAzureRMStorageShare_basic(rInt int, rString string, location string) string {
+	template := testAccAzureRMStorageShare_template(rInt, rString, location)
 	return fmt.Sprintf(`
-resource "azurerm_resource_group" "test" {
-  name     = "acctestRG-%[1]d"
-  location = "%[2]s"
+%s
+
+resource "azurerm_storage_share" "test" {
+  name                 = "testshare%s"
+  resource_group_name  = "${azurerm_resource_group.test.name}"
+  storage_account_name = "${azurerm_storage_account.test.name}"
+}
+`, template, rString)
 }
 
-resource "azurerm_storage_account" "test" {
-  name                     = "acctestacc%[3]s"
-  resource_group_name      = "${azurerm_resource_group.test.name}"
-  location                 = "${azurerm_resource_group.test.location}"
-  account_tier             = "Standard"
-  account_replication_type = "LRS"
+func testAccAzureRMStorageShare_metaData(rInt int, rString string, location string) string {
+	template := testAccAzureRMStorageShare_template(rInt, rString, location)
+	return fmt.Sprintf(`
+%s
 
-  tags = {
-    environment = "staging"
+resource "azurerm_storage_share" "test" {
+  name                 = "testshare%s"
+  resource_group_name  = "${azurerm_resource_group.test.name}"
+  storage_account_name = "${azurerm_storage_account.test.name}"
+
+  metadata {
+    hello = "world"
```

Comment:
Should we be checking case here?
```suggestion
    HellO = "WorlD!"
```

Reply:
nope - as in the other PR the casing on the keys is important (needs to be lower-case) - since this affects both I'll send a PR to fix that separately